### PR TITLE
[maven] Bump maven wrapper to 3.3.2

### PR DESCRIPTION
**Pull request by hazendaz**
Maven wrapper application is performed via

```
mvn -N org.apache.maven.plugins:maven-wrapper-plugin:3.2.0:wrapper -Dtype=source
```

At the moment, maven is still using legacy http url for apache license so I further run a git bash alias to do above plus the following.

```
sed -i 's,http://www.apache.org/licenses/LICENSE-2.0,https://www.apache.org/licenses/LICENSE-2.0,g'
```

So we do not promote http usage there.  I'm raising a PR to maven in that regard.  Will also be raising other items related to maven wrapper but note that in reality it takes maven years sometimes to apply simple things.  Last PR I had to them took nearly a year to get merged and released.

### CODE REVIEW

File: .mvn/wrapper/MavenWrapperDownloader.java [hunk 1]
```diff
@@ -1,117 +1,162 @@
 /*
- * Copyright 2007-present the original author or authors.
+ * Licensed to the Apache Software Foundation (ASF) under one
```
Comment (ThrawnCA) [inline .mvn/wrapper/MavenWrapperDownloader.java:2]: I don't think this is accurate. fb-contrib isn't being licensed to the Apache foundation, it's just using their licence boilerplate.

**Comment by hazendaz** [comment]
@ThrawnCA Maven licenses the files now, this is a script (ie maven wrapper plugin with type source), it comes with the license headers as indicated as files are not owned by individual repos but by th originating source (ie maven).  This changed over time as this was not originally a maven project, but apache took it and maven now owns it since 3.0.0.

File: .mvn/wrapper/MavenWrapperDownloader.java (hunk 1 above)
Comment (ThrawnCA) [reply .mvn/wrapper/MavenWrapperDownloader.java:2]: Ah, sorry, I misunderstood; this is specifically for the Maven wrapper, not for fb-contrib.

**Review by ThrawnCA — APPROVED** [review]

**Comment by hazendaz** [comment]
@mebigfatguy may as well merge this, 3.2.0 is in voting so to keep diff slightly cleaner, probably best to apply this as-is then pick that up afterwards.

File: .mvn/wrapper/MavenWrapperDownloader.java [hunk 2]
```diff
@@ -1,117 +1,162 @@
...
+import java.nio.file.LinkOption;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.nio.file.StandardOpenOption;
 import java.util.Properties;
 
-public class MavenWrapperDownloader {
+public final class MavenWrapperDownloader
+{
```
Comment (ThrawnCA) [inline .mvn/wrapper/MavenWrapperDownloader.java]: Putting braces on the next line is not generally the preferred style for Java, eg https://google.github.io/styleguide/javaguide.html#s4.1.2-blocks-k-r-style

Comment (hazendaz) [reply .mvn/wrapper/MavenWrapperDownloader.java]: These files are provided by maven, any issues with them need raised with the maven-wrapper project.

**Comment by hazendaz** [comment]
Re-based this PR, changed PR message and commit message and now to latest maven wrapper 3.2.0.

**Comment by hazendaz** [comment]
Note: these files all come from maven.  Only adjustment I've done is to use https on their license so we are not promoting redirects to apache license.  Application otherwise is a maven plugin run of maven wrapper.

```
mvn -N org.apache.maven.plugins:maven-wrapper-plugin:3.2.0:wrapper -Dtype=source
```

**Review by ThrawnCA — APPROVED** [review]

**Comment by hazendaz** [comment]
@mebigfatguy Can we get this merged?  I'll follow up with more as its out of date again but this is at least getting closer...

**Comment by hazendaz** [comment]
Re-updated to 3.3.2

**Review by ThrawnCA — APPROVED** [review]

**Pull request closed without merging**